pkg/resolution/resolver/git/resolver.go

```diff
@@ -180,6 +182,33 @@ func ResolveAnonymousGit(ctx context.Context, params map[string]string) (framewo
 	cloneOpts := &git.CloneOptions{
 		URL: repo,
 	}
+
+	secretRef := &secretCacheKey{
+		name: params[GitTokenParam],
+		key:  params[GitTokenKeyParam],
+	}
+	if secretRef.name != "" {
+		if secretRef.key == "" {
+			secretRef.key = DefaultTokenKeyParam
+		}
+		secretRef.ns = common.RequestNamespace(ctx)
+	} else {
+		secretRef = nil
+	}
+
+	auth := plumbTransport.AuthMethod(nil)
+	if secretRef != nil {
+		gitToken, err := getAPIToken(ctx, secretRef, kubeclient, logger, cache, ttl, params, GitTokenKeyParam)
+		if err != nil {
+			return nil, err
+		}
+		auth = &http.BasicAuth{
+			Username: "git",
```

Comment:
`git` is not guaranteed is it ? It is for `github` and `gitlab`, but one could use this with another provider that doesn't use `git` user but something else ? Should we make this configurable as well ?

Reply:
good idea but this could be a further improvement next, most forge provider would accept git